app/src/main/java/woowacourse/shopping/data/repository/ProductRepositoryImpl.kt

```
@@ -1,29 +1,74 @@
+            .url("${MockServer.getBaseUrl()}products")
+            .build()
+
+        try {
+            client.newCall(request).execute().use { response ->
+                if (!response.isSuccessful) {
+                    return@withContext emptyList()
+                }
+                val body = response.body?.string() ?: return@withContext emptyList()
+                json.decodeFromString<List<Product>>(body)
+            }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)